app/src/main/java/com/spascoding/taskycourse/feature_register_screen/presentation/login/LoginViewModel.kt

```diff
@@ -52,8 +51,8 @@ class LoginViewModel @Inject constructor(
         }
     }
 
-    fun validEmail(): Boolean = AuthPattern.EMAIL(state.email)
-    fun validPassword(): Boolean = AuthPattern.PASSWORD(state.password)
+    fun validEmail(): Boolean = AuthPattern.EMAIL(state.value.email)
+    fun validPassword(): Boolean = AuthPattern.PASSWORD(state.value.password)
```

Comment:
It's also unusual to have function names in caps, makes me think it's a constant. Why not something like `AuthPattern.isValidPassword(...)`?